numpy/testing/_private/utils.py

```diff
@@ -1949,8 +1949,10 @@ def assert_warns(warning_class, *args, **kwargs):
     >>> ret = np.testing.assert_warns(DeprecationWarning, deprecated_func, 4)
     >>> assert ret == 16
     """
-    if not args:
+    if not args and not kwargs:
         return _assert_warns_context(warning_class)
+    elif len(args) < 1:
+        raise RuntimeError("assert_warns called without args")
```

Comment:
Other libraries might have done the same thing we did in SciPy, thinking that `assert_warns` processed `match` like `pytest.warns`. If so, they may be confused when they start getting this error. Not sure what the best thing to do is, but one possibility is to check whether `'match' in kwargs`, and it so give a little more information.

Reply:
I improved the error handling and the test